dhall-lsp-server/src/LSP/Dispatcher.hs

```diff
@@ -109,10 +117,37 @@ dispatcher lf inp = do
           formattedDocument <- formatDocument uri tabSize insertSpaces
           publish req RspDocumentFormatting formattedDocument
 
+      ReqHover req -> do
+          liftIO $ LSP.Utility.logm "****** reactor: processing ReqHover"
+          let (J.Position line col) = req ^. (J.params . J.position)
+              doc = req ^. (J.params . J.textDocument . J.uri)
+              Just fileName = J.uriToFilePath doc
```

Comment:
I try to avoid non-exhaustive pattern match.  If you need a quick solution, you can do:

```haskell
fileName <- case J.uriToFilePath doc of
    Nothing       -> fail "A terse error message is fine for now"
    Just fileName -> return fileName
```